### ISSUE TITLE
Add Fitbounds support

### ISSUE THREAD

**Issue by timlogemann** [alex3165/react-mapbox-gl#78]
example: https://www.mapbox.com/mapbox-gl-js/example/fitbounds/
api: https://www.mapbox.com/mapbox-gl-js/api/#Map#fitBounds

In the mapbox example fitbounds only happens on the click of a button, however it would be logical to be able to do this on render when using dynamically generated markers and you want to display all markers on the map wherever they are.

Behavior would be similar to maxBounds but without the constraints in panning. Usage could also be similar by feeding the property and array of all marker coordinates, however it might simply make more sense to have it work with a boolean and having the map calculate the area based on all generated markers.

The difference between this and manually calculating the center based on all markers would be that it's much easier to use and also automatically sets the zoom to the correct value so that all markers fit into the viewable area.

**Comment by alex3165** [comment]
That's actually a good idea and it is an interesting and valuable feature to implement. However I would stick to the same API as mapbox-gl and pass an array of coordinates instead of a boolean because sometimes you want to be specific on which marker you want to display, so what about something we could use like this ?
```
const positions = [
  [1, -1],
  [2, -2],
  [3, -3],
  [4, -4]
]

const Comp = () => (
  <Map
    fitBounds={positions} // could be: positions.filter((el, i) => i < 2); to center on the 3 first markers only
  >
    <Layer>
      {
        positions.map(coord => <Feature coordinates={coord}/>)
      }
    </Layer>
  </Map>
);
```

**Comment by timlogemann** [comment]
Sticking with the same API is a good thing to do. I was thinking about my specific use case with the boolean. Being able to just give a small selection of markers instead of all of them makes this implementation more flexible.
I'm just diving into react so I'm not sure how much of a help I can be.

**Comment by alex3165** [comment]
@timlogemann Ready for your review : https://github.com/alex3165/react-mapbox-gl/pull/80

**Comment by timlogemann** [comment]
Looks good. I think we're ready to merge.

**Comment by alex3165** [comment]
Published : `react-mapbox-gl@0.21.0`
